mutt_socket.c

```diff
@@ -423,11 +418,6 @@ int raw_socket_write (CONNECTION* conn, const char* buf, size_t count)
     mutt_error (_("Error talking to %s (%s)"), conn->account.host,
 		strerror (errno));
     mutt_sleep (2);
-  } else if (errno == EINTR) {
```

Comment:
to catch if there was an error writing to the socket, one can simply compare `rc` with `count`, if they differ then the connection has ended abnormally, maybe as a result of a sigint, which can be checked with the presence of `SigInt` being set, something like:

```
if (rc != count)
{
  if (SigInt)
  {
    mutt_error (_("Connection to %s has been aborted"));
    SigInt = 0;
  }
  else
  {
    mutt_error (_("Unknown connection error while sending data to %s."), conn->account.host);
  }
  mutt_sleep (2);
}
```

of course that would need to be tested, especially with a `C-c` abort and forcing another issue. Not being an user of the IMAP features, I'm not sure how to create a testcase to check that code.